Flapjack/CoreData/CoreDataSource.swift

```diff
@@ -313,9 +319,17 @@ public class CoreDataSource<T: NSManagedObject & DataObject>: NSObject, NSFetche
     private func contextWillBeDestroyed(_ notification: Notification) {
         guard controller.managedObjectContext === notification.object as? NSManagedObjectContext else { return }
 
+        // Flag ourselves as existing under a zombie context (one about to be fully banished) so
+        //   that when we invoke an on-change block, any objects listening to us that _ask_ us for
+        //   our number of objects or our array of objects will get back empty ones (so as to
+        //   preserve data source functional integrity for collection/table view data sources).
+        // If an object needs to get a specific element from us based on these indices, that object
+        //   can still get it before it fully goes away using `object(at:)`.
```

Comment:
That's not a risky thing to do? It's a shame we can't do what the single core data source does and just zero out our array and count at this point, but since we defer to the controller for everything, it would add an unwanted level of indirection/caching in this object.  Could the `sectionInfo(for:)` check the zombie flag too, or is this behavior we *want* to allow?

Reply:
It's behavior we _want_ to allow because it's still safe. We have a class that listens for these change notifications and upon _deletion_ events, it uses the indices it's given to ask the data source for specific objects, so that it can remove those specific objects from its internally-held array.

Honestly, keeping the objects in the `fetchedObjects` array shouldn't be a risky thing to do either, since technically the objects are still around and the context hasn't been deleted yet, but collection / table views are quite finicky around this.